src/astro/files/locations/base.py

```diff
@@ -0,0 +1,52 @@
+import glob
+import os
+from abc import ABC, abstractmethod
+from typing import List, Optional
+from urllib.parse import urlparse
+
+
+class Location(ABC):
+    """Base Location abstract class"""
+
+    def __init__(self, path: str, conn_id: Optional[str] = None):
+        """
+        Location class constructor
+        :param path: Path to a file in the filesystem/Object stores
```

Comment:
Sphinx needs a blank-line between class description and param strings. Also let's move docstrings from __init__ to Class method itself. We don't need to but we can standardize it

```suggestion
        Location class constructor

        :param path: Path to a file in the filesystem/Object stores
```